### PR TITLE
Remove explicit add of publishinstallersandchecksums

### PR DESCRIPTION
**Pull request by chcosta**
See https://github.com/dotnet/windowsdesktop/pull/1078 for details,this PR is backporting that change to WindowsDesktop.

### CODE REVIEW

File: eng/common/post-build/publish-using-darc.ps1 [hunk 1]
```diff
@@ -29,7 +29,7 @@ try {
     $optionalParams.Add("--no-wait") | Out-Null
   }
 
-  if ("true" -eq $PublishInstallersAndChecksums) {
+  if ("false" -ne $PublishInstallersAndChecksums) {
```
Comment (jcagme) [inline eng/common/post-build/publish-using-darc.ps1:32]: Should we instead always add `--publish-installers-and-checksums` with no condition?

Comment (chcosta) [reply eng/common/post-build/publish-using-darc.ps1:32]: No idea.  This implementation would give us an opt out.  I'm not sure if it's necessary or not. Seems reasonable to opt out to me, do you have a strong argument against?

Comment (jcagme) [reply eng/common/post-build/publish-using-darc.ps1:32]: No, not really, I just thought based on what @epananth described the idea was to remove that option altogether. I'm fine if we leave this change as you have it as well

**Review by jcagme — APPROVED** [review]

File: eng/common/post-build/publish-using-darc.ps1 (hunk 1 above)
Comment (chcosta) [reply eng/common/post-build/publish-using-darc.ps1:32]: @epananth, merging this.  Please feel free to submit another PR if you feel there's a better fix.

File: eng/common/post-build/publish-using-darc.ps1 [hunk 2]
```diff
@@ -55,7 +55,6 @@ try {
   --publishing-infra-version $PublishingInfraVersion `
   --default-channels `
   --source-branch master `
-  --publish-installers-and-checksums `
```
Comment (riarenas) [inline eng/common/post-build/publish-using-darc.ps1:58]: This parameter is still needed. I'm reverting this change because it's breaking the build.

Comment (riarenas) [reply eng/common/post-build/publish-using-darc.ps1:58]: See https://dnceng.visualstudio.com/internal/_build/results?buildId=799558&view=results

**Comment by chcosta** [comment]
@riarenas @jcagme 

The parameter should still be included, unless someone explicitly sets it to false.  I think that reverting this is the wrong thing to do.

That build, is explicitly setting `/p:PublishInstallersAndChecksums=False`, and then expecting the parameter to be there.  That seems wrong.  The correct thing to do would be to remove that from the job - https://github.com/dotnet/arcade/blob/master/eng/promote-build.yml#L36-L39

Reverting this, is just lying about what you want done.  Also, this will have downstream consequences and break windowsdesktop now.  Please revert the revert.

**Comment by chcosta** [comment]
I'll prepare a change

**Comment by riarenas** [comment]
Reverting the revert results in official build being broken. Policy is to revert any changes which cause build breaks, such as this did. I think you're right on  what should happen, but I disagree with reverting this revert, and instead any new PRs should include the right fix so arcade doesn't remain broken. If windowsdesktop has made changes to eng/common that will be overridden by an arcade update, then I think the arcade update shouldn't be taken in that repo.

**Pull request merged**